### PR TITLE
arch: Avoid deprecated compiler flags

### PR DESCRIPTION
**Pull request by FabioLuporini**
recent NVC has deprecated gpu=pinned and suggested to rather use what this patch uses

### CODE REVIEW

**Review by georgebisbas — COMMENTED** [review]
WOuld this still work w previous nvc versions I guess?

File: devito/arch/compiler.py [hunk 1]
```diff
@@ -601,7 +601,7 @@ def __init_finalize__(self, **kwargs):
         platform = kwargs.pop('platform', configuration['platform'])
 
         if platform is NVIDIAX:
-            self.cflags.append('-gpu=pinned')
+            self.cflags.append('-gpu=mem:separate:pinnedalloc')
```
Comment (mloubout) [inline devito/arch/compiler.py:604]: probably need a version check like icx for openmp

**Review by EdCaunt — APPROVED** [review]

File: devito/arch/compiler.py (hunk 1 above)
Comment (FabioLuporini) [reply devito/arch/compiler.py]: (@georgebisbas ) pushed tweak so that older versions are guaranteed to work

**Review by georgebisbas — APPROVED** [review]
Thanks!

**Review by mloubout — APPROVED** [review]

**Pull request merged**